### PR TITLE
Resource aggregation worksheets

### PR DESCRIPTION
**Pull request by smartspot2**
Added support for multiple worksheets/solutions:
- Added a `Worksheet` model, with a foreign key to a `Resource` object
- Added ability to delete individual uploaded files and worksheets

### CODE REVIEW

File: csm_web/frontend/src/components/resource_aggregation/ResourceTable.tsx [hunk 1]
```diff
@@ -17,21 +17,64 @@ export const ResourceTable = ({ courseID }) => {
...
+   */
+  function getResourceFormData(
+    newResource: Resource,
+    fileFormDataMap: Map<number, Worksheet>,
+    newWorksheets: Array<Worksheet>
+  ) {
     let resourceFormData = new FormData();
     for (const [key, value] of Object.entries(newResource)) {
-      resourceFormData.set(key, value as any);
+      if (key != "worksheets") {
```
Comment (noortor) [inline csm_web/frontend/src/components/resource_aggregation/ResourceTable.tsx:50]: change != to !== for string comparison

File: csm_web/frontend/src/components/resource_aggregation/ResourceTable.tsx [hunk 2]
```diff
@@ -17,21 +17,64 @@ export const ResourceTable = ({ courseID }) => {
...
+        resourceFormData.set(key, value as any);
+      }
     }
-    for (let entry of fileFormData.entries()) {
-      resourceFormData.set(entry[0], entry[1]);
+    let idx = 0; // cumulative ndex in worksheet array
```
Comment (noortor) [inline csm_web/frontend/src/components/resource_aggregation/ResourceTable.tsx:54]: typo

Comment (smartspot2) [reply csm_web/frontend/src/components/resource_aggregation/ResourceTable.tsx]: Fixed!

File: csm_web/frontend/src/components/resource_aggregation/ResourceTable.tsx (hunk 1 above)
Comment (smartspot2) [reply csm_web/frontend/src/components/resource_aggregation/ResourceTable.tsx]: Fixed!

**Pull request merged**